internal/fs/buffer/inmemory_write_buffer_test.go

```diff
@@ -41,24 +56,371 @@ func (t *MemoryBufferTest) SetUp(ti *TestInfo) {
+	var err error
+	var sizeInMB int64 = 0
+	t.mb, err = CreateInMemoryWriteBuffer(uint(sizeInMB))
+
+	AssertNe(nil, err)
+	AssertTrue(strings.Contains(err.Error(), ZeroSizeBufferError))
+	AssertEq(nil, t.mb)
+}
+
+func (t *MemoryBufferTest) TestWritingToUninitializedBuffer() {
+	// Write to buffer
+	err := t.mb.WriteAt([]byte(smallContent1), 0)
+
+	AssertNe(nil, err)
+	AssertTrue(strings.Contains(err.Error(), ZeroSizeBufferError))
+}
+
+func (t *MemoryBufferTest) TestEnsureCurrentBuffer() {
+	t.createAndValidateInMemoryBuffer(bufferSizeInMB)
+
+	err := t.mb.ensureCurrentBuffer()
+
+	AssertEq(nil, err)
+	AssertEq(bufferSizeInMB*MiB, t.mb.bufferSize)
+	t.validateInMemoryBuffer(t.mb.current, bufferSizeInMB*MiB, 0, 0, bufferSizeInMB*MiB)
+}
+
+func (t *MemoryBufferTest) TestEnsureFlushedBuffer() {
+	t.createAndValidateInMemoryBuffer(bufferSizeInMB)
+
+	err := t.mb.ensureFlushedBuffer()
+
+	AssertEq(nil, err)
+	AssertEq(bufferSizeInMB*MiB, t.mb.bufferSize)
+	t.validateInMemoryBuffer(t.mb.flushed, bufferSizeInMB*MiB, 0, 0, 0)
+}
+
+func (t *MemoryBufferTest) TestEmptyWriteToInMemoryBuffer() {
+	t.createAndValidateInMemoryBuffer(bufferSizeInMB)
+
+	// Write to buffer
+	err := t.mb.WriteAt([]byte(""), 3)
+
+	AssertEq(nil, err)
+	AssertEq(0, t.mb.fileSize)
+	t.validateInMemoryBuffer(t.mb.current, 0, 0, 0, 0)
+	t.validateInMemoryBuffer(t.mb.flushed, 0, 0, 0, 0)
+}
+
+func (t *MemoryBufferTest) TestSingleWriteToInMemoryBuffer() {
+	t.createAndValidateInMemoryBuffer(bufferSizeInMB)
+
+	// Write to buffer
+	err := t.mb.WriteAt([]byte(smallContent1), 0)
+
+	AssertEq(nil, err)
+	AssertEq(smallContent1Size, t.mb.fileSize)
+	AssertTrue(bytes.Equal([]byte(smallContent1), t.mb.current.buffer[0:t.mb.fileSize]))
+	t.validateInMemoryBuffer(t.mb.current, bufferSizeInMB*MiB, 0, 0, bufferSizeInMB*MiB)
+	t.validateInMemoryBuffer(t.mb.flushed, 0, 0, 0, 0)
+}
+
+func (t *MemoryBufferTest) TestMultipleSequentialWritesToInMemoryBuffer() {
+	t.createAndValidateInMemoryBuffer(bufferSizeInMB)
+
+	// Write to buffer
+	err := t.mb.WriteAt([]byte(smallContent1), 0)
+	AssertEq(nil, err)
+	AssertEq(t.mb.fileSize, smallContent1Size)
+	err = t.mb.WriteAt([]byte(smallContent2), smallContent1Size)
+	AssertEq(nil, err)
+	AssertEq(t.mb.fileSize, smallContent1Size+smallContent2Size)
+	err = t.mb.WriteAt([]byte(smallContent3), smallContent1Size+smallContent2Size)
+	AssertEq(nil, err)
+	AssertEq(t.mb.fileSize, smallContent1Size+smallContent2Size+smallContent3Size)
+
+	AssertTrue(bytes.Equal([]byte(smallContent1+smallContent2+smallContent3), t.mb.current.buffer[0:t.mb.fileSize]))
+	t.validateInMemoryBuffer(t.mb.current, bufferSizeInMB*MiB, 0, 0, bufferSizeInMB*MiB)
+	t.validateInMemoryBuffer(t.mb.flushed, 0, 0, 0, 0)
+}
+
+func (t *MemoryBufferTest) Test2MBWriteTo2MBInMemoryBuffer() {
+	t.createAndValidateInMemoryBuffer(bufferSizeInMB)
+	data1 := t.generateRandomData(MiB)
+	data2 := t.generateRandomData(MiB)
+
+	// Write to buffer
+	err := t.mb.WriteAt(data1, 0)
+	AssertEq(nil, err)
```

Comment:
Please leave line gaps between WriteAt() calls and corresponding Assert statements.

Reply:
Not making this change to stick with the format of 
setup 
<line break>
test case
<line break>
Assertions

for the ease of reader.